core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -78,23 +76,23 @@ protected TableScan newRefinedScan(TableOperations ops, Table table, Schema sche
   }
 
   @Override
-  public CloseableIterable<FileScanTask> planFiles(TableOperations ops, Snapshot snapshot,
-                                                   Expression rowFilter, boolean ignoreResiduals,
-                                                   boolean caseSensitive, boolean colStats) {
-    ManifestGroup manifestGroup = new ManifestGroup(ops.io(), snapshot.dataManifests(), snapshot.deleteManifests())
-        .caseSensitive(caseSensitive)
-        .select(colStats ? SCAN_WITH_STATS_COLUMNS : SCAN_COLUMNS)
-        .filterData(rowFilter)
-        .specsById(ops.current().specsById())
+  public CloseableIterable<FileScanTask> doPlanFiles() {
+    Snapshot snapshot = snapshot();
+
+    ManifestGroup manifestGroup = new ManifestGroup(table().io(), snapshot.dataManifests(), snapshot.deleteManifests())
+        .caseSensitive(isCaseSensitive())
+        .select(colStats() ? SCAN_WITH_STATS_COLUMNS : SCAN_COLUMNS)
+        .filterData(filter())
+        .specsById(table().specs())
         .ignoreDeleted();
 
-    if (ignoreResiduals) {
+    if (shouldIgnoreResiduals()) {
```

Comment:
Nit: may rename `shouldIgnoreResiduals()` to `ignoreResiduals()` to be consistent.

Reply:
I cann't because of a conflicting setter method discussed [here](https://github.com/apache/iceberg/pull/4566#discussion_r855626731).